src/main/java/cz/smarteon/loxone/discovery/MiniserverDiscoverer.java

```diff
@@ -0,0 +1,180 @@
+package cz.smarteon.loxone.discovery;
+
+import org.java_websocket.util.NamedThreadFactory;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.TestOnly;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+import java.net.UnknownHostException;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Consumer;
+
+/**
+ * Provides Loxone miniserver discovery, using UDP.
+ * Can throw {@link LoxoneDiscoveryException} if something doesn't work correctly.
+ */
+public class MiniserverDiscoverer {
+
+    private static final byte[] DISCOVERY_PACKET_PAYLOAD = new byte[] { 0x00 };
+    private static final int REQUEST_PORT = 7070;
+    private static final int RESPONSE_PORT = 7071;
+
+    private final ExecutorService executor;
+    private final InetAddress broadcastAddress;
+    private int requestPort = REQUEST_PORT;
+    private int responsePort = RESPONSE_PORT;
+
+
+    /**
+     * Creates new instance, uses internal executor for UDP listener.
+     */
+    public MiniserverDiscoverer() {
+        this(null);
```

Comment:
Tohle je nějaký zvláštní. Proč je instance ExecutorService vytvářena až v Discover?

Reply:
no jinak bys nepoznal jesti je ten executor interní nebo externí, buď to bude takto a nebo tam přidám ještě boolean proměnnou vedle